tidy3d/plugins/mode/mode_solver.py

```diff
@@ -236,37 +374,41 @@ def to_source(
 
         Returns
         -------
-        ModeSource
-            Modal source containing specification in ``mode``.
+        :class:`.ModeSource`
+            Mode source with specifications taken from the ModeSolver instance and the method
+            inputs.
         """
 
-        center = self.plane.center
-        size = self.plane.size
         return ModeSource(
-            center=center,
-            size=size,
+            center=self.plane.center,
+            size=self.plane.size,
             source_time=source_time,
-            mode_spec=mode_spec,
+            mode_spec=self.mode_spec,
             mode_index=mode_index,
             direction=direction,
         )
 
-    def to_monitor(self, mode_spec: ModeSpec, freqs: List[float], name: str) -> ModeMonitor:
+    def to_monitor(self, freqs: List[float], name: str) -> ModeMonitor:
         """Creates :class:`ModeMonitor` from a ModeSolver instance + additional specifications.
 
         Parameters
         ----------
-        mode_spec : :class:`ModeSpec`
-            :class:`ModeSpec` object containing specifications of mode.
         freqs : List[float]
```

Comment:
these freqs are all stored in the ModeSolver?  So do we want to:
a) keep it the way it is
b) use `self.freqs` here?
b) use `self.freqs` if `freqs is None` or something fancier like this?

Reply:
Good question, I kinda like keeping it the way it is because it makes sure the user knows what's happening and passes the frequencies they want, or in general knows that it's possible to export a monitor with different (e.g. more) frequencies that they used in the solver. So I'm between a) or c), but like a).

Reply:
ok let's keep it (a) then, that's fine with me.